cmd/ledgerconfig/deletecmd/deletecmd.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package deletecmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/fabric-cli/pkg/environment"
+	"github.com/hyperledger/fabric-sdk-go/pkg/client/channel"
+	"github.com/spf13/cobra"
+	"github.com/trustbloc/fabric-cli-ext/cmd/ledgerconfig/common"
+)
+
+const (
+	use      = "delete"
+	desc     = "Delete ledger configuration"
+	longDesc = `
+The delete command allows the client to delete the MSP's configuration using search criteria. The criteria consists of:
+
+* MspID (mandatory)           - The MSP ID of the organization
+* PeerID (optional)           - The ID of the peer
+* AppName (optional)          - The application name
+* AppVersion (optional)       - The application version
+* ComponentName (optional)    - The component name
+* ComponentVersion (optional) - The component version
+
+Criteria may be specified as a JSON string (using the --criteria option) or it may be specified using the options:
+	--mspid, --peerid, --appname, --appver, --componentname and --componentver
+
+If PeerID and AppName are not specified then all of the MSP's configuration is deleted.
+`
+	examples = `
+- Delete an application's configuration for a given peer:
+    $ ./fabric ledgerconfig delete --mspid Org1MSP --peerid peer0.org1.com --appname myapp --appver 1
+
+- Delete a specific version of a component's configuration':
+    $ ./fabric ledgerconfig delete --mspid Org1MSP --appname myapp --appver 1 --componentname comp1 --componentver 1
+
+- Delete all configuration in Org1MSP:
+    $ ./fabric ledgerconfig delete --mspid Org1MSP
+`
+)
+
+const (
+	noPromptFlag        = "noprompt"
+	noPromptDescription = "If specified then delete operation will not prompt for confirmation. Example: --noprompt"
+
+	msgConfigDeleted   = "Configuration successfully deleted!"
+	msgAborted         = "Operation aborted"
+	msgContinueOrAbort = "Enter Y to continue or N to abort "
+	msgNoConfig        = "No configuration matches the given criteria"
+)
+
+// New creates a new delete command
+func New(settings *environment.Settings) *cobra.Command {
+	return newCmd(settings, nil)
+}
+
+func newCmd(settings *environment.Settings, p common.FactoryProvider) *cobra.Command {
+	c := &command{}
+
+	cmd := &cobra.Command{
+		Use:     use,
+		Short:   desc,
```

Comment:
Suggestion: add `SuggestFor` for `remove` (unless that will be its own command, which would be a good source of confusion).

Reply:
Note, this isn't blocking, I've approved.